pkg/sync/syncWorker.go

```diff
@@ -0,0 +1,205 @@
+package sync
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/xmtp/xmtpd/pkg/db/queries"
+	"github.com/xmtp/xmtpd/pkg/proto/xmtpv4/message_api"
+	"github.com/xmtp/xmtpd/pkg/registrant"
+	"github.com/xmtp/xmtpd/pkg/registry"
+	"github.com/xmtp/xmtpd/pkg/tracing"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/protobuf/proto"
+)
+
+type syncWorker struct {
+	ctx          context.Context
+	log          *zap.Logger
+	nodeRegistry registry.NodeRegistry
+	registrant   *registrant.Registrant
+	store        *sql.DB
+	wg           sync.WaitGroup
+}
+
+func startSyncWorker(
+	ctx context.Context,
+	log *zap.Logger,
+	nodeRegistry registry.NodeRegistry,
+	registrant *registrant.Registrant,
+	store *sql.DB,
+) (*syncWorker, error) {
+	s := &syncWorker{
+		ctx:          ctx,
+		log:          log.With(zap.String("method", "syncWorker")),
+		nodeRegistry: nodeRegistry,
+		registrant:   registrant,
+		store:        store,
+		wg:           sync.WaitGroup{},
+	}
+	if err := s.start(); err != nil {
+		return nil, err
+	}
+	return s, nil
+}
+
+func (s *syncWorker) start() error {
+	nodes, err := s.nodeRegistry.GetNodes()
+	if err != nil {
+		return err
+	}
+	for _, node := range nodes {
+		if node.NodeID == s.registrant.NodeID() || !node.IsHealthy {
```

Comment:
you also want to skip `NodeID == 0` and `!node.IsValid`

Reply:
I don't think the node registry will ever have NodeID < 100, so we shouldn't need to check for it - that was just a workaround we had earlier.

Added a check for `node.IsValidConfig`, that seems to be new - @neekolas why is this needed? Would it make sense to collapse it into `node.IsHealthy`?

Reply:
the current registry is still full of invalid <100 nodes. Luckily that should be caught by the validity check.